addOns/encoder/src/main/java/org/zaproxy/addon/encoder/processors/predefined/FullHtmlStringEncoder.java

```diff
@@ -0,0 +1,32 @@
+/*
+ * Zed Attack Proxy (ZAP) and its related class files.
+ *
+ * ZAP is an HTTP/HTTPS proxy for assessing web application security.
+ *
+ * Copyright 2022 The ZAP Development Team
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.zaproxy.addon.encoder.processors.predefined;
+
+public class FullHtmlStringEncoder extends DefaultEncodeDecodeProcessor {
+
+    @Override
+    protected String processInternal(String value) {
+        StringBuilder output = new StringBuilder();
```

Comment:
- The initial size could be calculated to an approximate final value.
- Worth adding a test, even if there's not much to test it ensures it stays working as expected.

Reply:
You mean set size as something like value.length x 5 (since most code points are likely 2 digits at least, plus the 3 we add each char)

Reply:
Yes.

Reply:
Okay